src/percy-agent-client/percy-agent-client.ts

```diff
@@ -9,28 +9,46 @@ export class PercyAgentClient {
     this.healthCheck()
   }
 
-  post(path: string, data: any) {
+  post(path: string, data: any): Promise <any> {
     if (!this.agentConnected) {
       console.warn('percy agent not started.')
-      return
+      return Promise.resolve()
     }
 
-    this.xhr.open('post', `${this.agentHost}${path}`, false) // synchronous request
-    this.xhr.setRequestHeader('Content-Type', 'application/json')
-    this.xhr.send(JSON.stringify(data))
+    return this._makeRequest(
+      `${this.agentHost}${path}`,
+      'post',
+      JSON.stringify(data),
+      'application/json',
+    )
   }
 
-  healthCheck() {
-    try {
-      this.xhr.open('get', `${this.agentHost}/percy/healthcheck`, false)
-      this.xhr.onload = () => {
-        if (this.xhr.status === 200) {
-          this.agentConnected = true
+  healthCheck(): Promise <any> {
+    return this._makeRequest(
+      `${this.agentHost}/percy/healthcheck`, 'get')
+    .then(() => {
+      this.agentConnected = true
```

Comment:
It looks like you handled an error coming back previously, with `this.agentConnected = false`. This is no longer present because there's no `catch`.